packages/editor-ui/src/utils/objectUtils.ts

```diff
@@ -18,3 +18,48 @@ export const searchInObject = (obj: ObjectOrArray, searchString: string): boolea
 			? searchInObject(entry, searchString)
 			: entry?.toString().toLowerCase().includes(searchString.toLowerCase()),
 	);
+
+/**
+ * Efficiently and deeply compare two objects. Handles nested objects and arrays.
+ * Returns true if they are equal, false otherwise.
+ */
+export function deepCompare<T extends object>(
```

Comment:
Do we need to implement custom method for this? 🤔 We're already using `isEqual` from `lodash-es` in our codebase which also does a deep comparison

Reply:
I reinvented the wheel 😅... Thanks for pointing it out. Removed the helper.